include/swift/SIL/SILBuilder.h

```diff
@@ -712,6 +712,23 @@ class SILBuilder {
                       BeginBorrowInst(getSILDebugLocation(Loc), LV));
   }
 
+  /// If \p v is compatible with guaranteed, return an empty
+  /// SILValue. Otherwise, borrow \p v and return the borrowed value. If
+  /// ownership is disabled in this function, then the original value is
+  /// returned.
+  ///
+  /// This allows for recursive code that deals with borrows to only borrow the
+  /// first time the function is called. It also by returning v when ownership
+  /// is disabled makes it easy to write code that works on ossa and on non-ossa
+  /// SIL.
+  SILValue emitBeginBorrowOperation(SILLocation loc, SILValue v) {
+    if (!hasOwnership())
+      return v;
+    if (v.getOwnershipKind().isCompatibleWith(ValueOwnershipKind::Guaranteed))
+      return SILValue();
```

Comment:
Why is this not returning v?

Reply:
It is in the PR description I think but should be commented in the code - much easier than doing git blame: "if the value is already guaranteed, return an empty SILValue(). This
is useful since when writing a recursive function it enables one to tell if
one needs to insert an end_borrow or not."

Reply:
Hmmm... I was looking at this and I discovered that this was actually a thinko in my code that I fixed in a subsequent commit locally. Let me fix this up.